components/module-components/infinitie-carousel/InfiniteCarouselData.ts

```diff
@@ -0,0 +1,30 @@
+const InfiniteCarouselData = [
+  {
+    text: 'UI/UX Design',
+  },
+  {
+    text: 'Frontend Development',
+  },
+  {
+    text: 'Backend Development',
+  },
+  {
+    text: 'Quality Assurance',
+  },
+  {
+    text: 'Marketing',
+  },
+  {
+    text: 'DevOps',
+  },
+  {
+    text: 'Product Management',
+  },
+];
+
+const repeatCount = 4;
+
```

Comment:
you don't necessarily need this structure of an array of objects
You can create an array of just string values. Then you can then use Array with fill() and flat().
(this is more efficient because we wont be creating new arrays through the spread operator: Array.from({ length: repeatCount }, () => [...InfiniteCarouselData]).flat() ).

Reply:
If you want to keep the structure above, we need to keep in mind how objects are stored.
If we do this multipliedArray[0].text = 'Design'; 
Array.fill()  will make the change in every set because both 'UI/UX' refer to the same object in memory.
Array.from() creates shallow copies, so each repetition in the array will contain a shallow copy of the object, meaning if we do multipliedArray[0].text = 'Design' it will only change the first instance.

Reply:
The data in the array is immutable, meaning we don't want a scenario where the array with the repeated elements is [  {text: 'UI/UX Design',},{text: 'Frontend Development',},{text: 'Backend Development',},{text: 'Quality Assurance',}, {text: 'Marketing',}, {text: 'DevOps',},{text: 'Product Management',},  **{text: 'Design',},**text: 'Frontend Development',},{text: 'Backend Development',},{text: 'Quality Assurance',}, {text: 'Marketing',}, {text: 'DevOps',},{text: 'Product Management',},]

Reply:
you can leave it as it, just wanted to share another perspective ✌️

Reply:
I made the change, you are right. It is the better solution.